hoomd/mpcd/RejectionVirtualParticleFillerGPU.cuh

```diff
@@ -0,0 +1,265 @@
+                                       const Scalar vel_factor,
+                                       const unsigned int filler_id,
+                                       const unsigned int type,
+                                       const unsigned int N_virt_max,
+                                       const unsigned int timestep,
+                                       const unsigned int seed,
+                                       const unsigned int block_size,
+                                       const Geometry geom)
+    {
+    // one thread per particle
+    const unsigned int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if (idx >= N_virt_max)
+        return;
+
+    // initialize random number generator for positions and velocity
+    hoomd::RandomGenerator rng(hoomd::RNGIdentifier::RejectionFiller, seed, timestep, first_tag+idx, filler_id);
+    d_tmp_pos[idx] = make_scalar4(hoomd::UniformDistribution<Scalar>(lo.x, hi.x)(rng),
+                                  hoomd::UniformDistribution<Scalar>(lo.y, hi.y)(rng),
+                                  hoomd::UniformDistribution<Scalar>(lo.z, hi.z)(rng),
+                                  __int_as_scalar(type));
+
+    hoomd::NormalDistribution<Scalar> gen(vel_factor, 0.0);
+    Scalar3 vel;
+    gen(vel.x, vel.y, rng);
+    vel.z = gen(rng);
+    d_tmp_vel[idx] = make_scalar4(vel.x,
+                                  vel.y,
+                                  vel.z,
+                                  __int_as_scalar(mpcd::detail::NO_CELL));
+
+    // check if particle is inside/outside the confining geometry
+    d_track_bounded_particles[idx] = geom.isOutside(make_scalar3(d_tmp_pos[idx].x,
+                                                                 d_tmp_pos[idx].y,
+                                                                 d_tmp_pos[idx].z));
+    }
+
+/*!
+ * \b implementation
+ * Using one thread per particle, we assign the particle position, velocity and tags using the compacted indices
+ * array as an input.
+ */
+__global__ void parallel_copy(unsigned int *d_compact_indices,
+                              Scalar4 *d_permanent_positions,
+                              Scalar4 *d_permanent_velocities,
+                              unsigned int *d_permanent_tags,
+                              Scalar4 *d_temporary_positions,
+                              Scalar4 *d_temporary_velocities,
+                              const unsigned int first_idx,
+                              const unsigned int first_tag,
+                              const unsigned int n_virtual,
+                              const unsigned int block_size)
+    {
+    // one thread per virtual particle
+    const unsigned int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if (idx >= n_virtual)
+        return;
+
+    // d_compact_indices holds accepted particle indices from the temporary arrays
+    unsigned int particleIdx = d_compact_indices[idx];
```

Comment:
Mark as const, and suggest not to use camel case for consistency with rest of code
```suggestion
    const unsigned int pidx = d_compact_indices[idx];
```